Java/src/com/couchbase/litecore/C4Database.java

```diff
@@ -170,13 +178,20 @@ public C4Document getBySequence(long sequence) throws LiteCoreException {
 
     // - Purging and Expiration
 
-    public void setExpiration(String docID, long timestamp)
+    public boolean setExpiration(String docID, Long timestamp)
             throws LiteCoreException {
-        C4Document.setExpiration(handle, docID, timestamp);
+        if(timestamp == null)
+            return C4Document.setExpiration(handle, docID, 0);
+        else
+            return C4Document.setExpiration(handle, docID, timestamp);
     }
 
-    public long getExpiration(String docID) {
-        return C4Document.getExpiration(handle, docID);
+    public Long getExpiration(String docID) {
```

Comment:
The signature should be `public long getExpiration(String docID)`.

Reply:
There is a reason I am using Long here since we are returning null when there is no expiration is set for the document.

Reply:
We would like to make C4Database close to LiteCore implementation as much as possible. Returning null will be done on CouchbaseLite side.